src/main/java/com/example/resourcemanagerapp/controller/MyResourceController.java

```diff
@@ -0,0 +1,85 @@
+package com.example.resourcemanagerapp.controller;
+
+
+import com.example.resourcemanagerapp.additionalTypes.EnumChecker;
+import com.example.resourcemanagerapp.additionalTypes.MyResourceType;
+import com.example.resourcemanagerapp.service.MyResourceServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.configurationprocessor.json.JSONException;
+import org.springframework.boot.configurationprocessor.json.JSONObject;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+public class MyResourceController {
+
+    private final MyResourceServiceImpl myResourceService;
+
+    @Autowired
+    public MyResourceController(MyResourceServiceImpl myResourceService) {
+        this.myResourceService = myResourceService;
+    }
+
+    @PostMapping(value = "/resource/add")
+    public ResponseEntity addResource(@RequestParam String name, @RequestParam Integer userId, @RequestParam String type,
```

Comment:
zamiast sprawdzac czy jest taki enum - EnumCheckerem, moznaby poprostu przyjac enuma zamiast Stringa, Spring autmatycznie to przemapuje, a jesli takiego nie bedzie to rzuci bledem. Pozwoli to usunac enumCheckera